magento/inventory
# MSI-2031 in-store pickup attribute

This is still WIP and based on the non-merged branch 2035. Please feel free to review the latest three commits and provide feedback for me to make changes while I continue with the ToDo items.

### Description
This PR adds an extension attribute to the Sources to define if the source supports in-store pickup. UI Components have been created to view/filter on the Source listing grid and edit on the Source form.

### Fixed Issues (if relevant)
1. magento-engcom/msi#2031: Add possibility to define Source to use for In Store Pickup

### Manual testing scenarios
1. Create a Source and notice on the form editing page the ability to set if in-store pickup is available.
2. Save the Source and notice the attribute has persisted in the database table inventory_source.
3. View Sources in the listing grid and change the visibility for the in-store pickup attribute and you will be able to see the status of the in-store pickup per source in the grid.
4. Filter the listing grid on the status of the in-store pickup and the sources should be filtered appropriately.

### ToDo in this PR or another where appropriate:
1. Add in ability to toggle extension attribute via API.
2. Create automated tests.
